mac/setup/build.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -e
+set -u
+
+# Include our resource functions; they're pretty useful!
+. ../../resources/shellHelperFunctions.sh
+
+# Please note that this build script (understandably) assumes that it is running on Mac OS X.
+verify_on_mac
+
+#
+# Constants
+#
+
+SOURCEAPP="./Install Keyman.app"
+TARGETPATH=../Keyman4MacIM/output
+TARGETAPP="$TARGETPATH/Install Keyman.app"
+KEYMANAPP=../Keyman4MacIM/build/Release/Keyman.app
+
+#
+# Build textinputsource
+#
+
+pushd textinputsource
+./build.sh
+popd
+
+#
+# Build install candidate of "Install Keyman"
+#
+
+# TODO: get script folder so we can run this from anywhere safely
+# TODO: set version number in install image
+
+# Remove old install image
+mkdir -p "$TARGETPATH"
+rm -rf "$TARGETAPP" || true
+
+# Copy new install image to output foldercer
+cp -R "$SOURCEAPP" "$TARGETPATH/"
+
+# Copy Keyman.app into install image
+cp -R "$KEYMANAPP" "$TARGETAPP/Contents/MacOS/Keyman.app"
+
+# Copy textinputsource into install image
+cp textinputsource/textinputsource "$TARGETAPP/Contents/MacOS/textinputsource"
+
+# NOW TO notarize...
+xattr -rc "$TARGETAPP"
+codesign --force --options runtime --deep --sign "${CERTIFICATE_ID}" "$TARGETAPP"
+
+#
+# Notarize the app (copied from ../build.sh)
+# TODO: merge notarization into function
+#
+
+TARGET_ZIP_PATH="$TARGETPATH/Install Keyman.zip"
+TARGET_APP_PATH="$TARGETAPP"
+ALTOOL_LOG_PATH="$TARGETPATH/altool.log"
+
+echo_heading "Zipping Install Keyman.app for notarization to $TARGET_ZIP_PATH"
+
+/usr/bin/ditto -c -k --keepParent "$TARGET_APP_PATH" "$TARGET_ZIP_PATH"
+
+echo_heading "Uploading Install Keyman.zip to Apple for notarization"
+
+xcrun altool --notarize-app --primary-bundle-id "com.Keyman.install.zip" --asc-provider "$APPSTORECONNECT_PROVIDER" --username "$APPSTORECONNECT_USERNAME" --password @env:APPSTORECONNECT_PASSWORD --file "$TARGET_ZIP_PATH" --output-format xml > $ALTOOL_LOG_PATH || fail "altool failed"
```

Comment:
Do you want to fail the build if `APPSTORECONNECT_*` credentials aren't provided?

Reply:
I think the build will fail anyway, won't it, with a relevant message? We've got `set -u` at the top of the script.